provision.tf

```diff
@@ -0,0 +1,28 @@
+# Ensure that the instance is reachable via `ssh` before deploying
+resource "null_resource" "wait" {
+  provisioner "remote-exec" {
+    connection {
+      host        = aws_instance.factorio.public_dns
+      private_key = tls_private_key.deploy-key.private_key_openssh
+    }
+
+    inline = [":"] # Do nothing; we're just testing SSH connectivity
+  }
+}
+
+locals {
+  # run the command in the nix dev shell
+  nix-shell-cmd = "nix --extra-experimental-features \"nix-command flakes\" develop .# --command"
```

Comment:
Is the `.#` directory dependent? I like to keep things runnable from any PWD if possible.

Reply:
Terraform will collapse the path to the path of the module, but you're right explicitly calling off `${path.module}` is more readable.